### PR TITLE
Sync fork repo before opening a translation PR

### LINKED ISSUE #8427: Form to submit translations via PR creation is not working
The form to submit translations and create a pull request hasn't been working for me (probably others since I don't see other translations coming in).
It was working around 3 weeks ago, despite having had some similar issues before, I'm only reporting it now because it happened again and there's more stuff to translate.

Here's the error report if you want to take a look:
Request ID: 0e0d4534-4315-4c73-bfe9-dcc682e42609

### PR DESCRIPTION
**Pull request by jonatanklosko**
Closes #8427.

Note that this requires "workflow" scope on the token, I edited the existing token to match that.

### CODE REVIEW

File: WcaOnRails/app/controllers/translations_controller.rb [hunk 1]
```diff
@@ -42,6 +42,14 @@ def update
...
     content_digest = Digest::SHA1.hexdigest(content)
     branch_name = "translation-#{locale}-#{content_digest}"
+    # We create a branch pointing to upstream master SHA. This stopped working
+    # unless the commit exists in the fork repository, so we always sync first.
+    # Octokit doesn't expost a method for this, so we make a regular request.
+    RestClient.post("https://api.github.com/repos/#{origin_repo}/merge-upstream", '{"branch":"master"}', {
+                      accept: "application/vnd.github+json",
+                      authorization: "Bearer #{AppSecrets.GITHUB_CREATE_PR_ACCESS_TOKEN}",
+                      x_github_api_version: "2022-11-28",
+                    })
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/translations_controller.rb:52]:
```suggestion
    # We create a branch pointing to upstream master SHA. This stopped working
    # unless the commit exists in the fork repository, so we always sync first.
    # Octokit doesn't expost a direct method for this, so we monkey-patch our request.
    Octokit.client.post "#{Octokit::Repository.path upstream_repo}/merge-upstream", { branch: 'master' }
```

Comment (gregorbg) [reply WcaOnRails/app/controllers/translations_controller.rb:52]: (FYI this `Octokit.client.post` approach with a string-templated URL based on `Octokit::Repository.path` is exactly how Octokit internally routes pretty much all API requests that it exposes convenience methods for)

File: WcaOnRails/app/controllers/translations_controller.rb [hunk 2]
```diff
@@ -42,6 +42,10 @@ def update
     message = "Update #{locale} translation."
     content_digest = Digest::SHA1.hexdigest(content)
     branch_name = "translation-#{locale}-#{content_digest}"
+    # We create a branch pointing to upstream master SHA. This stopped working
+    # unless the commit exists in the fork repository, so we always sync first.
+    # Octokit doesn't expost a direct method for this, so we monkey-patch our request.
+    Octokit.client.post "#{Octokit::Repository.path origin_repo}/merge-upstream", { branch: 'master' }
```
Comment (jonatanklosko) [inline WcaOnRails/app/controllers/translations_controller.rb:48]:
```suggestion
    # Octokit doesn't expose a direct method for this, so we make a custom request.
    Octokit.client.post("#{Octokit::Repository.path(origin_repo)}/merge-upstream", { branch: "master" })
```

**Comment by jonatanklosko** [comment]
Tested in the console, the request works as expected :)

**Review by gregorbg — APPROVED** [review]

**Pull request merged**